pinot-spi/src/main/java/org/apache/pinot/spi/config/table/UpsertConfig.java

```diff
@@ -40,9 +40,16 @@
     APPEND, INCREMENT, OVERWRITE, UNION
   }
 
+  public enum HashFunction {
+    NONE, MD5, MURMUR3
```

Comment:
Suggest using `null` to represent disabling hashing. We don't expect people to explicitly put `NONE` as the hash function.

Reply:
it's easier for reading the serialized value

Reply:
Why is that easier? You still need to perform `null` check right?
My concern is that:
1. `NONE` is not really a valid `HashFunction`
2. We usually use `null` to represent that a config is not available in most places. It might be a little bit confusing to use `NONE` just for upsert

Reply:
well, we have `NONE` as upsert mode value?

Reply:
reading is easier. When I look at the table config, I can view the full info, and dont have to infer if hash function is enabled or not

Reply:
That one I also suggested using `null` in the code review... Having an upsert config with `NONE` mode which disables it doesn't seems very clean to me. But I'm okay if you strongly feel this way is better